### PR TITLE
PYTHON-2715 Use hello command for monitoring when supported

### PR DESCRIPTION
**Pull request by prashantmital**
We should wait for https://github.com/mongodb/specifications/pull/990 to be merged before merging this but the code changes can be reviewed.

Note that I have not yet synced the `legacy_hello` tests as they seem to still be in flux.

### CODE REVIEW

File: pymongo/pool.py
Comment (ShaneHarvey) [inline pymongo/pool.py:576]: Suggest merging this with the server_api check above:
```
if self.opts.server_api or self.hello_ok:
...
```

File: pymongo/pool.py
Comment (ShaneHarvey) [inline pymongo/pool.py:577]: Is helloOk only supposed to be sent with the initial handshake? If so we should move it to `_ismaster()` under the `if performing_handshake` check.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Hard to tell from the PR UI but did you remove the old SDAM tests before resyncing? This makes sure we delete tests that are no longer in the specs repo.

File: pymongo/pool.py
Comment (prashantmital) [reply pymongo/pool.py]: Not needed anymore since `helloOk` is only sent on initial handshake.

File: pymongo/pool.py
Comment (prashantmital) [reply pymongo/pool.py]: Thanks for catching that. Done!

**Review by ShaneHarvey — APPROVED** [review]
LGTM

File: test/discovery_and_monitoring_integration/minPoolSize-error.json
Comment (ShaneHarvey) [inline test/discovery_and_monitoring_integration/minPoolSize-error.json:20]: This file needs to be reverted to fix the test failures.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

**Comment by ShaneHarvey** [comment]
Can you schedule all the "Versioned API" tests too?

File: test/discovery_and_monitoring_integration/minPoolSize-error.json
Comment (ShaneHarvey) [reply test/discovery_and_monitoring_integration/minPoolSize-error.json:20]: This file needs both isMaster and hello as mentioned in the spec PR.

**Comment by prashantmital** [comment]
@ShaneHarvey updated the spec test so this is ready for your LGTM. I will wait until the spec PR is merged before merging this.

**Review by ShaneHarvey — APPROVED** [review]
LGTM

File: pymongo/pool.py
Comment (ShaneHarvey) [reply pymongo/pool.py]: I brought up this issue in the spec PR and James says we should send helloOk on all isMaster commands, not just the initial handshake.

File: pymongo/pool.py
Comment (ShaneHarvey) [inline pymongo/pool.py:587]: The the Spec PR James says we should send helloOk on all isMaster commands, not just the initial handshake: https://github.com/mongodb/specifications/pull/990#issuecomment-870792241

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/pool.py
Comment (prashantmital) [reply pymongo/pool.py]: Done.

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**
